v1.15 Backports 2024-10-02

aanm opened the pull request:
Backporting all of these PRs to avoid backport conflicts and to optimize our CI infrastructure. PRs for the remaining stable branches will happen afterwards.

This will effectively remove Conformance E2E (ci-e2e) from our v1.15 branch.

* [x] #29459
 * [x] #34811 (@aanm) :warning: resolved conflicts
 * [x] #34742 (@aanm) :warning: resolved conflicts
 * [x] #34800 (@aanm) :warning: resolved conflicts
 * [x] #34845 (@aanm)
 * [x] #34843 (@aanm) :warning: resolved conflicts
 * [x] #34847 (@aanm) :warning: resolved conflicts
 * [x] #34848 (@aanm) :warning: resolved conflicts
 * [x] #34866 (@aanm) :warning: resolved conflicts
 * [x] #35088 (@aanm)
 * [x] #35141 (@aanm)
 * [x] #33451  (@aanm)
 * [x] #32528
 * [x] #35227

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
29459 34811 34742 34800 34845 34843 34847 34848 34866 35088 35141 33451 32528 35227
```

lmb committed:
bpf: do not invoke llc separately

[ upstream commit 49f571f0fbcdbdd6f5ec392e2546d32b4a7c53dc ]

Update the various bpf Makefiles to not invoke llc anymore. This matches
what the loader does now.

Signed-off-by: Lorenz Bauer <lmb@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

lmb committed:
tools/dev-doctor: do not check for llc

[ upstream commit ef54b49428b1bb8507385c87205815d117d34805 ]

Compiling bpf doesn't invoke llc directly anymore, stop checking for its presence.

Signed-off-by: Lorenz Bauer <lmb@isovalent.com>
Signed-off-by: André Martins <andre@cilium.io>

mhofstetter committed:
ci: increase wait duration after upgrade/downgrade in E2E upgrade test

[ upstream commit 7a930fa40204afb273e3823a6a6659d402f177a2 ]

E2E Upgrade tests randomly fail if health endpoints aren't available after
an upgrade within the given default timeout of 5min.

```
Errors:                cilium             cilium-c4qkb    controller cilium-health-ep is failing since 2m7s (15x): Get "http://10.244.2.186:4240/hello": dial tcp 10.244.2.186:4240: connect: no route to host
```

It looks like the health server unix listener isn't available at that time.

Therefore, this commit increases the timeout from 5 to 10 minutes.

Suggested-by: Marco Iorio <marco.iorio@isovalent.com>
Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

aanm committed:
.github/workflows: remove conformance-e2e

[ upstream commit 76adcd90d0076bdd477e0f8139c3c0fcf38da71c ]

The conformance-e2e test has the same coverage as tests-e2e-upgrade.
Although connectivity tests will not run on a "clean state" it does not
really matter given that if the tests fail on the upgrade tests then the
PR should not be merged anyway.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github/workflows: add skip-upgrade flag

[ upstream commit a7d60e1205a30039294b91876629668dcade4ced ]

When introducing new feature we might need to skip the "upgrade" path as
new features can't be "downgraded" to a previous version since they
don't exist. Thus, we are introducing a new flag: 'skip-upgrade'.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: fix non-existing env.cilium_stable_version

[ upstream commit d79ad46b6bd735ad8424c0193b8f9427bebb9681 ]

The env.cilium_stable_version does not exist, using
'steps.vars.outputs.downgrade_version' as it should have the same
effect.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: re-enable configurations in e2e-upgrade

[ upstream commit 46724c5e157a90333bf636a5c2f53670cb6898fc ]

Some tests got out of sync with conformance-e2e. As conformance-e2e got
removed from our CI, this commit is updating e2e-upgrade to have the
same configurations used in the conformance-e2e.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: fix lvh-kind warnings

[ upstream commit 871decdebb5108e6d6b12ed7d11a79d0c0551035 ]

There are some warnings from the lvh-kind GitHub action. This commit
fixes them.

Also, add renovate configuration to update the lvh-kind action version
automatically.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: install golang action after checkout

[ upstream commit 0883901cd41a7bb7731fc69e6617513c0b5a3b5f ]

The Golang GitHub Action relies on a cache keyed by the `go.sum` file.
If the repository is not checked out, the action can't access the cache.
To resolve this, we will move the repository checkout step before the
Golang setup.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: specify cache-dependency-path in lint-workflows

[ upstream commit 307751b9547e8b747f253489622141aec12d247b ]

The Golang GitHub Action relies on a cache keyed by the `go.sum` file.
If the repository is not checked out in the default directory, the
action can't access the cache.
To resolve this, we will set the right directory with the
'cache-dependency-path' flag to point to the directory that contains the
checked out source code on the previous step.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: use caches for lint-build-commits

[ upstream commit 12b480f043e3660841553d70822d0ee2ce17cde2 ]

With theses changes, the workflow run time will decreases tremendously
up with the usage of ccache and golang caches.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
images/Dockerfile: re-order mount commands

[ upstream commit 60735161e2b973fa4a70db1c043f7f9bb1336f0c ]

Split mount commands into multiple lines for easy readability.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: simplify cache usage in docker images

[ upstream commit c6b999b708630cc48f407be16c658880d7ba5688 ]

This commit introduces the usage of Golang caches to speed up the docker
images used in our CI.

It will use a dedicated GH workflow to build Golang caches once a commit
is pushed into main. Then, PRs can re-use this cache to speed up the
image build process.

As an example, cilium docker images build time was reduced from ~16m to
~6m38s.
The availability of cilium docker image amd64, used in our tests, was
reduced from 439 seconds to 232 seconds.

The overall CI run for all images was reduced from ~16m20s to ~7m3s.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: recreate caches on push to main

[ upstream commit b73cce34ea6b35cbdf33086e6e02080db8be7112 ]

Using push to main events to recreate the caches will allow branches /
PRs to reuse those caches when building the code. As the diff between
the main branch and the PR is minimal, the caches will speed up the
building process.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: simplify build-images-ci workflow

[ upstream commit 5ce311271620e630d360fbca2faae6245230def0 ]

Now that we build golang caches outside of this workflow, we can
simplify this workflow by reusing code for both main and PRs.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: create cache directories on cache miss

[ upstream commit f5c49944c0c0d772492ae71ae6dd2c993fc3926a ]

ON a cache miss, the cache directories don't exist which makes the CI
build image to fail. Adding this missing step to recreate the cache
directories on cache misses will prevent the CI build from failing.

Fixes: c6b999b70863 (".github: simplify cache usage in docker images")
Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: clean up disk for lint-build workflow

[ upstream commit 3c2df71539af4eee9f7e435b2cf4c2f6d5e5a36d ]

This workflow is running out of disk so we should run the cleanup script
to make more space available.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github/workflows: change GH runners

GitHub now uses 4 CPU cores for default runners, up from 2. This allows
us to use these runners instead of self-hosted ones, benefiting from
features like nested virtualization.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: decrease conformance-aks schedule to 8 hours

To avoid waiting CPU resources, we should run the tests every 8 hours
instead of 6.

Signed-off-by: André Martins <andre@cilium.io>

julianwiedmann commented on .github/workflows/tests-e2e-upgrade.yaml:
```diff
@@ -186,6 +183,7 @@ jobs:
             kpr: 'false'
             tunnel: 'geneve'
             endpoint-routes: 'true'
+            misc: 'socketLB.enabled=false,nodePort.enabled=true,bpf.masquerade=true'
```
This doesn't exist in the `v1.15` version of the conformance-e2e workflow.

julianwiedmann commented on .github/workflows/tests-e2e-upgrade.yaml:
```diff
@@ -243,6 +241,14 @@ jobs:
 
+          - name: '13'
+            # renovate: datasource=docker depName=quay.io/lvh-images/kind
+            kernel: 'rhel8-20240404.144247'
```
In the `v1.15` version of the conformance-e2e workflow this is still `4.19-20240710.064909`.

julianwiedmann commented on .github/workflows/conformance-e2e.yaml:
```diff
@@ -1,474 +0,0 @@
-            kube-proxy: 'iptables'
-            kpr: 'false'
-            tunnel: 'vxlan'
-            host-fw: 'true'
```
To note - in the `tests-e2e-upgrade` workflow the `host-fw` option is set to false (to work-around downgrade bugs). So if we now solely use that workflow for testing, we're effectively losing coverage.

aanm replied:
@julianwiedmann thanks for the review. I've copied the host-fw's specific tests intot he tests-e2e-upgrade tests and skip the upgrade for those configurations. We shouldn't be losing any coverage now.

julianwiedmann reviewed (changes requested):
Some minor changes needed.

aanm committed:
.github: do not push floating tag from PRs

[ upstream commit 63fe139e599d216f4b5a5ddd6f7879057af34b0c ]

With 5ce311271620, PRs started to push their changes into the 'latest'
tag which would cause CI issues. With this commit we make sure the
'latest' tag only happens on "push" events for the default branch of the
repository.

Fixes: 5ce311271620 (".github: simplify build-images-ci workflow")
Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: push floating tag for push events for stable branches

Fixes: 8b3fe57af911 (".github: do not push floating tag from PRs")
Signed-off-by: André Martins <andre@cilium.io>

joestringer reviewed (approved):
I glanced through most of the `.github` files and nothing obvious jumped out at me. Broadly speaking it is very difficult to give a good review on what is effectively a rewrite of significant portions of the workflows on an old branch.

julianwiedmann reviewed (approved):
thank you!
